packages/terra-worklist-data-grid/src/WorklistDataGrid.jsx

```diff
@@ -77,25 +104,181 @@ function WorklistDataGrid(props) {
+      }
+    } else if (onDisableSelectableRows) {
+      ariaLiveMsg.current = intl.formatMessage({ id: 'Terra.worklist-data-grid.row-selection-mode-disabled' });
+      onDisableSelectableRows();
+    }
+  };
+
+  const selectRows = (selectAllRows, rowId, rowIndex) => {
+    let rowLabel;
+    let msgId = 'Terra.worklist-data-grid.all-rows-selected';
+    if (!selectAllRows) {
+      const isSelectAction = !rows[rowIndex - 1].isSelected; // Determine if this is select or unselected.
+      msgId = isSelectAction ? 'Terra.worklist-data-grid.row-selection-template' : 'Terra.worklist-data-grid.row-selection-cleared-template';
+      rowLabel = rows[rowIndex - 1].ariaLabel || rowIndex;
+    }
+    ariaLiveMsg.current = intl.formatMessage({ id: msgId }, { row: rowLabel });
+    if (selectAllRows && onRowSelectAll) {
+      onRowSelectAll();
+    } else if (onRowSelect) {
+      onRowSelect(rowId);
+    }
+  };
+
+  const handleMoveCellFocus = (fromCell, toCell) => {
+    removeTabStop(fromCell.row, fromCell.col);
+    setFocusedRowCol(toCell.row, toCell.col, true);
+  };
+
+  const handleCellSelectionChange = (rowId, columnId, cellCoordinates) => {
+    if (!hasSelectableRows) {
+      ariaLiveMsg.current = intl.formatMessage({
+        id: rowId ? 'Terra.worklist-data-grid.cell-selection-template' : 'Terra.worklist-data-grid.cell-selection-cleared',
+      }, { row: cellCoordinates.row, column: cellCoordinates.col });
+    }
+    setFocusedRowCol(cellCoordinates.row, cellCoordinates.col, true);
+    if ((rowId !== currentSelectedCell?.rowId) || (columnId !== currentSelectedCell?.columnId)) {
+      setCurrentSelectedCell((rowId && columnId) ? { rowId, columnId } : null);
+    }
+  };
+
+  const mapGridCellToDataCell = (cellGridCoordinates) => (
+    // The grid has an additional row for the heading and
+    // may have an additional column when when selection is active.
+    { row: cellGridCoordinates.row - 1, col: cellGridCoordinates.col + (hasSelectableRows ? -1 : 0) }
+  );
+
+  const selectCell = (cellRowIdColId, cellGridCoordinates) => {
+    // If current cell is selected, do nothing.
+    if (isCellSelected(cellGridCoordinates.row, cellGridCoordinates.col)) {
+      return;
+    }
+    // Determine if the cell selection should proceed.
+    const cellDataCoordinates = mapGridCellToDataCell(cellGridCoordinates);
+    const cell = rows[cellDataCoordinates.row].cells[cellDataCoordinates.col];
+    if ((cell.isSelectable === false) || cell.isMasked) {
+      return;
+    }
+    // Make note of cell that is currently selected.
+    handleCellSelectionChange(cellRowIdColId.rowId, cellRowIdColId.columnId, cellGridCoordinates);
+    if (onCellSelect) {
+      onCellSelect(cellRowIdColId.rowId, cellRowIdColId.columnId);
+    }
+  };
+
+  const handleColumnSelect = (columnId, cellCoordinates) => {
+    handleCellSelectionChange(null, null, cellCoordinates);
+    if (onColumnSelect && !(hasSelectableRows && cellCoordinates.col === 0)) {
```

Comment:
Shouldn't the row selection column just not be selectable?

Reply:
We handled this before the isSelectable was merged in. We can now use the isSelectable that we have actually added the ROW_SELECTION_COLUMN constant containing that column's properties. I will make note and in the row selection jira, update the check to use the isSelectable.

Reply:
Actually, it is and the check could be updated now that the isSelectable code is merged in. Incidentally, we already added the field to the ROW_SELECTION_COLUMN constant so using it here will be straight forward. I am however suggesting that we do that minor tweak in the row selection Jira.